pkg/apis/metal3/v1alpha1/baremetalhost_types.go

```diff
@@ -160,6 +169,9 @@ type BareMetalHostSpec struct {
 	// +kubebuilder:validation:Pattern=`[0-9a-fA-F]{2}(:[0-9a-fA-F]{2}){5}`
 	BootMACAddress string `json:"bootMACAddress,omitempty"`
 
+	// Boot mode for baremetal server
+	BootMode BootMode `json:"bootMode,omitempty"`
```

Comment:
We should configure the kubebuilder validation for this enum in the same way that we do elsewhere in this file.